### PR TITLE
fix(curriculum): No longer use global variables

### LINKED ISSUE #50887: Test Relies On Variable In User Editable Code
### Describe the Issue

The tests for this problem rely upon a variable in the user editable code. If the users delete this variable, then a correct solution will not pass the tests. We should update the tests to not do this.

### Affected Page

https://www.freecodecamp.org/learn/project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i

### Your code

```javascript
function maximumPathSumI(triangle) {
  // Copy the triangle
  //   Note: not needed if we are ok with mutating the input
  const triangleCopy = triangle.map(row => [...row]);
  // Find max path
  for (let i = 1; i < triangleCopy.length; i++) {
    triangleCopy[i][0] += triangleCopy[i - 1][0];
    for (let j = 1; j < i; j++) {
      triangleCopy[i][j] += Math.max(triangleCopy[i - 1][j - 1], triangleCopy[i - 1][j]);
    }
    triangleCopy[i][i] += triangleCopy[i - 1][i - 1];
  }
  return Math.max(...triangleCopy[triangleCopy.length - 1]);
}

// NOTE: global variable deleted from seed code here
```

### Expected behavior

Tests should be isolated.

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #50887

I first changed the test text to explicitly mention the argument passed to the function where I could. For Problem 67, this was problematic as best with the 100 rows. I then moved the definition of `testTriangle` into the background right next to `numTriangle` and made sure to remove the variable from the user code. I believe this could work.

### CODE REVIEW

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md [hunk 1]
```diff
@@ -41,19 +41,19 @@ Find the maximum total from top to bottom of the triangle below:
...
 
-`maximumPathSumI(testTriangle)` should return 23.
+`maximumPathSumI([[3, 0, 0, 0], [7, 4, 0, 0],[2, 4, 6, 0],[8, 5, 9, 3]])` should return 23.
 
 ```js
 assert.strictEqual(maximumPathSumI(testTriangle), 23);
 ```
 
-`maximumPathSumI(numTriangle)` should return 1074.
+`maximumPathSumI([[75, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [95, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [17, 47, 82, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [18, 35, 87, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [20, 4, 82, 47, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [19, 1, 23, 75, 3, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0], [88, 2, 77, 73, 7, 63, 67, 0, 0, 0, 0, 0, 0, 0, 0], [99, 65, 4, 28, 6, 16, 70, 92, 0, 0, 0, 0, 0, 0, 0], [41, 41, 26, 56, 83, 40, 80, 70, 33, 0, 0, 0, 0, 0, 0], [41, 48, 72, 33, 47, 32, 37, 16, 94, 29, 0, 0, 0, 0, 0], [53, 71, 44, 65, 25, 43, 91, 52, 97, 51, 14, 0, 0, 0, 0], [70, 11, 33, 28, 77, 73, 17, 78, 39, 68, 17, 57, 0, 0, 0], [91, 71, 52, 38, 17, 14, 91, 43, 58, 50, 27, 29, 48, 0, 0], [63, 66, 4, 68, 89, 53, 67, 30, 73, 16, 69, 87, 40, 31, 0], [4, 62, 98, 27, 23, 9, 70, 98, 73, 93, 38, 53, 60, 4, 23]])` should return 1074.
```
Comment (jeremylt) [inline curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:56]: I'm honestly not sure if we want to spell this one out or not. I like being explicit about the test cases, but  this is kinda hard to read.

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-67-maximum-path-sum-ii.md [hunk 2]
```diff
@@ -63,11 +63,6 @@ function maximumPathSumII(triangle) {
   return true;
 }
 
-const testTriangle = [[3, 0, 0, 0],
-                      [7, 4, 0, 0],
-                      [2, 4, 6, 0],
-                      [8, 5, 9, 3]];
-
 maximumPathSumII(testTriangle);
```
Comment (jeremylt) [inline curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-67-maximum-path-sum-ii.md:66]: I think this needs to be removed too?

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md [hunk 3]
```diff
@@ -65,6 +65,8 @@ assert.strictEqual(maximumPathSumI(numTriangle), 1074);
 
 ```js
 const numTriangle = [[75, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [95, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [17, 47, 82, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [18, 35, 87, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [20, 4, 82, 47, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [19, 1, 23, 75, 3, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0], [88, 2, 77, 73, 7, 63, 67, 0, 0, 0, 0, 0, 0, 0, 0], [99, 65, 4, 28, 6, 16, 70, 92, 0, 0, 0, 0, 0, 0, 0], [41, 41, 26, 56, 83, 40, 80, 70, 33, 0, 0, 0, 0, 0, 0], [41, 48, 72, 33, 47, 32, 37, 16, 94, 29, 0, 0, 0, 0, 0], [53, 71, 44, 65, 25, 43, 91, 52, 97, 51, 14, 0, 0, 0, 0], [70, 11, 33, 28, 77, 73, 17, 78, 39, 68, 17, 57, 0, 0, 0], [91, 71, 52, 38, 17, 14, 91, 43, 58, 50, 27, 29, 48, 0, 0], [63, 66, 4, 68, 89, 53, 67, 30, 73, 16, 69, 87, 40, 31, 0], [4, 62, 98, 27, 23, 9, 70, 98, 73, 93, 38, 53, 60, 4, 23]];
+
+const testTriangle = [[3, 0, 0, 0], [7, 4, 0, 0],[2, 4, 6, 0],[8, 5, 9, 3]];
```
Comment (jeremylt) [inline curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:69]: Should we do this, just so we don't collide with likely user test variable names?

```suggestion
const _numTriangle = [[75, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [95, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [17, 47, 82, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [18, 35, 87, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [20, 4, 82, 47, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [19, 1, 23, 75, 3, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0], [88, 2, 77, 73, 7, 63, 67, 0, 0, 0, 0, 0, 0, 0, 0], [99, 65, 4, 28, 6, 16, 70, 92, 0, 0, 0, 0, 0, 0, 0], [41, 41, 26, 56, 83, 40, 80, 70, 33, 0, 0, 0, 0, 0, 0], [41, 48, 72, 33, 47, 32, 37, 16, 94, 29, 0, 0, 0, 0, 0], [53, 71, 44, 65, 25, 43, 91, 52, 97, 51, 14, 0, 0, 0, 0], [70, 11, 33, 28, 77, 73, 17, 78, 39, 68, 17, 57, 0, 0, 0], [91, 71, 52, 38, 17, 14, 91, 43, 58, 50, 27, 29, 48, 0, 0], [63, 66, 4, 68, 89, 53, 67, 30, 73, 16, 69, 87, 40, 31, 0], [4, 62, 98, 27, 23, 9, 70, 98, 73, 93, 38, 53, 60, 4, 23]];

const _testTriangle = [[3, 0, 0, 0], [7, 4, 0, 0],[2, 4, 6, 0],[8, 5, 9, 3]];
```

Comment (naomi-lgbt) [reply curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:69]: Yes, this is what we typically try to do.

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md (hunk 1 above)
Comment (a2937) [reply curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:56]: Yeah it was exceedingly large when I tested it.

**Comment by moT01** [comment]
Looks like there's some failing tests. I added the `waiting update` label until those are fixed 👍

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md [hunk 4]
```diff
@@ -41,19 +41,19 @@ Find the maximum total from top to bottom of the triangle below:
 
 # --hints--
 
-`maximumPathSumI(testTriangle)` should return a number.
+`maximumPathSumI([[3, 0, 0, 0], [7, 4, 0, 0],[2, 4, 6, 0],[8, 5, 9, 3]])` should return a number.
 
 ```js
 assert(typeof maximumPathSumI(testTriangle) === 'number');
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:47]:
```suggestion
assert(typeof maximumPathSumI(_testTriangle) === 'number');
```

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md [hunk 5]
```diff
@@ -41,19 +41,19 @@ Find the maximum total from top to bottom of the triangle below:
...
 ```
 
-`maximumPathSumI(testTriangle)` should return 23.
+`maximumPathSumI([[3, 0, 0, 0], [7, 4, 0, 0],[2, 4, 6, 0],[8, 5, 9, 3]])` should return 23.
 
 ```js
 assert.strictEqual(maximumPathSumI(testTriangle), 23);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:53]:
```suggestion
assert.strictEqual(maximumPathSumI(_testTriangle), 23);
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Looks like the tests just need an update with the new variable name

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md (hunk 1 above)
Comment (jeremylt) [reply curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:56]: I could go either way here. Is there a picture of the output we can discuss?

**Review by jeremylt — APPROVED** [review]

File: curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md (hunk 1 above)
Comment (a2937) [reply curriculum/challenges/english/18-project-euler/project-euler-problems-1-to-100/problem-18-maximum-path-sum-i.md:56]: Imgur image

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**